pkg/provisioner/controller.go

```diff
@@ -351,7 +365,15 @@ func (c *obcController) handleProvisionClaim(key string, obc *v1alpha1.ObjectBuc
 	if err != nil {
 		return fmt.Errorf("error %s bucket: %v", verb, err)
 	} else if ob == (&v1alpha1.ObjectBucket{}) {
-		return fmt.Errorf("provisioner returned nil/empty object bucket")
+		return fmt.Errorf("provisioner returned empty object bucket")
+	}
+
+	// Fill in known information in the object bucket struct as soon as possible
+	setBasicOBInfo()
```

Comment:
If the provisioner were to set ob.Spec.ReclaimPolicy, say based on some flag or config, we will overwrite that now.

Reply:
I didn't modify any of the functional code that set the spec, so this would have been the case all along. I will add another commit to fix that small issue here also.

Reply:
Looking into this further, I'm not sure this is an issue. Currently the reclaim policy is set here in the lib controller from the storage class. If the provisioner implementation sets its own reclaim policy that is different from the storage class, that to me seems like a bug. But perhaps there is a case I can't think of where Rook or another provisioner implementation would want to take override the reclaim policy. I think such a case might confuse users, however.

Reply:
I think it would be unusual for the provisioner to modify the OB, but it could and there might be a good reason to do so in certain cases. If  the lib wants to prohibit this then maybe it should call `setBasicOBInfo` after the provisioner returns?